service/service.go

```diff
@@ -146,16 +143,37 @@ func (svc *Service) HandleMessage(ctx context.Context, message kafka.Message) (s
 			return instanceID, err
 		}
 
-		for _, optionToPost := range lineDimensions {
-			if err := svc.DatasetClient.PostInstanceDimensions(ctx, svc.AuthToken, instanceID, optionToPost); err != nil {
-				log.Event(ctx, "encountered error sending request to datset api", log.ERROR, log.Error(err), log.Data{"instance_id": instanceID, "csv_line": line})
-				return instanceID, err
+		// loop through each dimension option identified in the CSV row
+		for optionKey, optionToPost := range lineDimensions {
+
+			// if the option is already in the full map of options then do not add it
+			if _, ok := dimensionOptions[optionKey]; ok {
+				continue
 			}
+
+			log.Event(ctx, "adding dimension to post", log.INFO, log.Data{"instance_id": instanceID, "dimension_option": optionKey})
```

Comment:
How many dimensions are there ?
Do you want to be logging every one ?
(and similarly for any other loops)

Reply:
Lots of dimensions - logging removed 😅